include/triton/core/tritonserver.h

```diff
@@ -1862,6 +1880,24 @@ TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_ServerModelIndex(
 TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_ServerLoadModel(
     TRITONSERVER_Server* server, const char* model_name);
 
+/// Load the requested model or reload the model if it is already
+/// loaded, with load parameters provided. The function does not return until
+/// the model is loaded or fails to load. Returned error indicates if model
+/// loaded successfully or not.
+/// Currently the below parameter names are recognized:
+/// - "config" : string parameter that contains a JSON representation of the
+/// model configuration. This config will be used for loading the model instead
+/// of the one in the model directory.
+///
+/// \param server The inference server object.
+/// \param model_name The name of the model.
+/// \param parameters The array of load parameters.
+/// \param parameter_count The number of parameters.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_ServerLoadModel(
```

Comment:
For C there is no name mangling, so need a different name for this function. Perhaps TRITONSERVER_ServerLoadModelWithParameters

Reply:
Thinking about possible future extension on loading multiple models at once, should we make this multi model (still only support single model currently) so we don't need to add another API when we add the support?

Reply:
Is it not sufficient to just call ServerLoadModel multiple times to load multiple models? I'd rather not add "multi" to all the load/unload protocol unless necessary